src/main/networking/vtunnel.ts

```diff
@@ -10,14 +10,26 @@ import * as childProcess from '@/utils/childProcess';
 import BackgroundProcess from '@/utils/backgroundProcess';
 
 const vtunnelConfig = 'vtunnel-config.yaml';
-const localHost = '127.0.0.1';
-const credServerVsockPort = 17361;
-const credServerHandshakePort = 17362;
-const credServerPeerPort = 3030;
-const credServerPort = '6109';
-
-export class vtunnel {
-  vsockProxy = new BackgroundProcess('Credentials Helper Host Proxy', {
+
+/**
+ * Configuration Object for Vtunnel Proxy.
+ */
+export interface VtunnelConfig {
+  name: string;
+  handshakePort: number;
+  vsockHostPort: number;
+  peerAddress: string;
+  peerPort: number;
+  upstreamServerAddress: string;
+}
+
+/**
+ * Vtunnel is a management and integration class for Vtunnel Proxy of Rancher Desktop.
+ */
+class Vtunnel {
```

Comment:
Naming suggestion (optional):
```suggestion
class VTunnel {
```